libraries/lib-stretching-sequence/ClipSegment.h

```diff
@@ -41,7 +41,9 @@ class STRETCHING_SEQUENCE_API ClipSegment final : public AudioSegment
    const sampleCount mTotalNumSamplesToProduce;
    sampleCount mTotalNumSamplesProduced = 0;
    ClipTimeAndPitchSource mSource;
-   // Careful that this guy is constructed last, as its ctor refers to *this.
+   std::mutex mStretcherMutex;
```

Comment:
It seems that you use the mutex for making parts of the TimeAndPitchInterface to be thread safe. I don't understand, why this mutex is not in the TimeAndPitchInterface implementation in this case.

Reply:
In `ClipSegment.cpp`, you can see that we add two callbacks for pitch shift change and formant preservation toggling. These callbacks are issued from the main thread, meanwhile the processing is on the audio thread.
Since we have the guarantee that the `StaffPadTimeAndPitch` ctor is completely executed before these callbacks are registered, it would be safe to internalize the mutex, but the safety is more straightforward to implement and for reviewers to verify this way.
The drawback with this approach is that we lock the mutex at frame rate also if there is no time stretching. I still haven't got to the bottom of whether this is a bad thing, and I was thinking maybe you could enlighten me. In our case there'd only be contention when the user changes pitch during playback.